webpack/assets/javascripts/react_app/components/common/Fill/FillReducer.js

```diff
@@ -0,0 +1,27 @@
+import Immutable from 'seamless-immutable';
+
+import { REGISTER_FILL, REMOVE_FILLED_COMPONENT } from './FillConstants';
+
+const initialState = Immutable({
+  RegisteredComponent: {},
+});
+
+export default (state = initialState, action) => {
+  const { payload } = action;
+
+  switch (action.type) {
+    case REGISTER_FILL:
+      return state.setIn(
+        ['RegisteredComponent', payload.slotId, payload.fillId],
+        true
+      );
+
+    case REMOVE_FILLED_COMPONENT:
+      return state.setIn(
+        ['RegisteredComponent', payload.slotId, payload.fillId],
```

Comment:
Same as [comment](https://github.com/Katello/katello/pull/8039#discussion_r268562069) on the other end.